src/sage/matrix/berlekamp_massey.py

```diff
@@ -72,6 +72,12 @@ def berlekamp_massey(a):
         Traceback (most recent call last):
         ...
         ValueError: argument must have an even number of terms
+
+    Check that :issue:`36172` is fixed::
+
+        sage: p = next_prime(2**64)
+        sage: ls = [GF(p).random_element() for _ in range(2000)]
+        sage: _ = berlekamp_massey(ls)
     """
```

Comment:
Really, there is no test in this doctest. Instead of this, how about reporting measured memory consumptions before and after this PR as a comment in the description of this PR. 

Anyway, the code looks good to me.

Reply:
Please also report on measured time consumptions.

Reply:
I know I should do that but I am not sure how to accurately do it on mac. [This](http://sporadic.stanford.edu/reference/misc/sage/misc/getusage.html) function `get_memory_usage()` seems to be gone in Sage now.

Reply:
Well... This may help

https://github.com/sagemath/sage/issues/33637#issuecomment-1418169379

Reply:
I have updated the PR description :) Thanks for the quick help

Reply:
Thanks for the quick action. Would you remove the useless doctest?